### PR TITLE
I18n 

### PR DESCRIPTION
**Pull request by Sakrecoer**
This PR enables the French translations we currently have and adds an explanation to the README.md file for translator wanting to enable a new language. I hope i'm doing this right (sorry for many minor commits)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,3 +25,29 @@ npm run build
...
+You are welcome to submit translations of the wiki pages. So far we only have a small portion of the wiki translated to French. We currently only have capacity to review the following languages:
+
+- French
+- German
+- Swedish
+- Turkish
+
+If your translation is a mere Google translation without any correction, the Pull Request will likely not be accepted.
+
+**Before you begin translating, please consider your submission as *commitment* to maintain the pages in sync with any changes happening in the english version**
```
Comment (nop33) [inline README.md]:
```suggestion
**Before you begin translating, please consider your submission as *commitment* to maintain the pages in sync with any changes happening in the English version**
```

File: README.md [hunk 2]
```diff
@@ -25,3 +25,29 @@ npm run build
...
+
+### Enable a new language for translation
+
+1. Create a folder for the documents inside the `i18n` folder. For example german: `mkdir -p i18n/<two-letter-international-code>/docusaurus-plugin-content-docs/current` (*note that you will need replace `<two-letter-international-code>` with the actual two-letter international code*).
```
Comment (nop33) [inline README.md]: Where can the translator find the list of `<two-letter-international-code>`?

File: README.md [hunk 3]
```diff
@@ -25,3 +25,29 @@ npm run build
...
+2. Copy all the content from `/docs` into the new folder: `cp -r docs i18n/<two-letter-international-code>/docusaurus-plugin-content-docs/current/` (*again, note that you will need replace `<two-letter-international-code>` with the actual two-letter international code*).
+3. Remove all the `_category_.json` files from the subfolders inside `i18n/<two-letter-international-code>/docusaurus-plugin-content-docs/current`.
+4. Build the translation files for the navbar and the footer: `npm run write-translations -- --locale <two-letter-international-code>`.
+
+Now you can start translating any and all file in `i18n/<two-letter-international-code>/docusaurus-plugin-content-docs/current`
```
Comment (nop33) [inline README.md]:
```suggestion
Now you can start translating all files in `i18n/<two-letter-international-code>/docusaurus-plugin-content-docs/current`
```

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+### wallet
+
+Vous pouvez télécharger le wallet de bureau à partir d'ici [GitHub](https://github.com/alephium/desktop-wallet/releases/latest).
+
+Alternativement, notre noeud complet possède un wallet intégré avec des fonctionnalités avancées, vous pouvez suivre notre [Guide du wallet](wallet/Wallet-Guide.md) pour apprendre à l'utiliser.
+
+[^1] : Le dossier d'accueil dépend de votre système : `C:\Users\<votre-username>` sous Windows, `/Users/<votre-username>` sous macOS, `/home/<votre-username>` sous Linux.
+
+[explorer] : https://explorer.alephium.org
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md]: The space before the colon breaks the link rendering:

image

```suggestion
[explorer]: https://explorer.alephium.org
```

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+Pour le tutoriel sur le minage, vous pouvez suivre notre [Guide du minage en solo](mining/Solo-Mining-Guide.md) ou [Guide du minage en pool](mining/Pool-Mining-Guide.md).
+
+### wallet
+
+Vous pouvez télécharger le wallet de bureau à partir d'ici [GitHub](https://github.com/alephium/desktop-wallet/releases/latest).
+
+Alternativement, notre noeud complet possède un wallet intégré avec des fonctionnalités avancées, vous pouvez suivre notre [Guide du wallet](wallet/Wallet-Guide.md) pour apprendre à l'utiliser.
+
+[^1] : Le dossier d'accueil dépend de votre système : `C:\Users\<votre-username>` sous Windows, `/Users/<votre-username>` sous macOS, `/home/<votre-username>` sous Linux.
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md]: Again, there should be no space before the colon.

```suggestion
[^1]: Le dossier d'accueil dépend de votre système : `C:\Users\<votre-username>` sous Windows, `/Users/<votre-username>` sous macOS, `/home/<votre-username>` sous Linux.
```

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-More.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+---
+sidebar_position : 40
+titre : Plus d'information et dépannage
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-More.md]: There should be no spaces before colons in this section. Please apply this in other pages as well

```suggestion
sidebar_position: 40
titre: Plus d'information et dépannage
```

File: i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
+sidebar_position : 20
+title : Sur les poussières d'output et l'explosion des états
+---
+
+
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md]: Too many empty lines

File: i18n/fr/docusaurus-plugin-content-docs/current/project/5min-overview.md [hunk 8]
```diff
@@ -0,0 +1,191 @@
...
+
+Voici notre dernière mise à jour de code : <https://github.com/alephium/alephium/releases/latest/>
+
+Venez commiter et faire une pull request avec nous : <https://github.com/alephium>
+
+- [Full node](https://github.com/alephium/alephium)
+- [Wallet](https://github.com/alephium/alephium-wallet/releases/latest)
+- [Explorateur](https://github.com/alephium/explorer)
+- [JS-SDK](https://github.com/alephium/js-sdk)
+- [Wiki][https://wiki.alephium.org]
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/project/5min-overview.md]:
```suggestion
- [Wiki](https://wiki.alephium.org)
```

File: i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+> ## ⚠️ Cette page n'a pas encore été traduite. Vous pouvez le faire en suivant le lien en pied de page.
+
+The wallet API can be called using our Swagger UI at [http://127.0.0.1:12973/docs](http://127.0.0.1:12973/docs) or by using `curl`. Make sure that your full node is running so you could access the Swagger UI.
+
+## Create a new wallet
+
+You can create a new wallet by doing a POST with the following data on `/wallets`.
+
+```json
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: If we change this to `js`, then the the code highlighter won't complain with a red background (json cannot have comments, js can.
```suggestion
```js
```

**Review by nop33 — COMMENTED** [review]
I noticed the following message 

image

However, when I click the link at the bottom of the page, I am redirected to the English version of the page in the `master` branch. Maybe this gets fixed once merged? :/

image

Moreover, I see that the images were copied too... That's unnecessary and a waste of space since the images don't change. Let's investigate how images can be reused from the English language.

Lastly, several changes happened on master since this PR was opened. I don't know what's the best way to track this. We need to establish some kind of process for keeping track of missing translations. When a PR to edit a page is created, we somehow need to track that this particular change needs to be reflected in the translations of this page.

**Comment by Sakrecoer** [comment]
right, feels like the edit-link should automatically direct to the correct language version. Is it happening also on the built version? (the test version can only display one language at a time.) I would imagine there's a setting for this as the URL of the edit link is built of the URL set in the config file...

On 6 June 2022 08:52:21 UTC, Ilias Trichopoulos ***@***.***> wrote:
***@***.*** commented on this pull request.

**Comment by nop33** [comment]
> right, feels like the edit-link should automatically direct to the correct language version. Is it happening also on the built version? (the test version can only display one language at a time.) I would imagine there's a setting for this as the URL of the edit link is built of the URL set in the config file...

I run the website with the instructions I saw in the README:

```
npm run build && npm run serve
```

File: README.md (hunk 3 above)
Comment (Sakrecoer) [reply README.md]: all AND any because you don't have to translate all of them if you only have caoacity for one

**Comment by Sakrecoer** [comment]
Thanks a lot for all the corrections, @nop33 ! 💜

**Comment by Sakrecoer** [comment]
> I run the website with the instructions I saw in the README:

yeah that builds it all (i got that command from you hehe)

File: i18n/fr/docusaurus-plugin-content-docs/current/dapps/A-Primer-With-The-Desktop-Wallet.md [hunk 10]
```diff
@@ -0,0 +1,101 @@
+---
+sidebar_position: 10
+title: A primer with the desktop wallet
+---
+
+> ## ⚠️ Cette page n'a pas encore été traduite. Vous pouvez le faire en suivant le lien en pied de page.
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/dapps/A-Primer-With-The-Desktop-Wallet.md]: There is a nicer way to display this warning messages. Check out this doc page! https://docusaurus.io/docs/markdown-features/admonitions

```suggestion
:::info
Cette page n'a pas encore été traduite. Vous pouvez le faire en suivant le lien en pied de page.
:::
```

image

File: README.md (hunk 2 above)
Comment (Sakrecoer) [reply README.md]: Good catch, thought i had included it:
https://www.loc.gov/standards/iso639-2/php/English_list.php

Should probably replace `two-letter-international-code>` with `<ISO-639-1>` also

File: i18n/fr/docusaurus-plugin-content-docs/current/dapps/A-Primer-With-The-Desktop-Wallet.md (hunk 10 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/dapps/A-Primer-With-The-Desktop-Wallet.md]: beautiful! thank you!!

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-More.md (hunk 6 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-More.md]: hehe! french habits (not mine 😅🙃)

**Comment by Sakrecoer** [comment]
This is how Facebook handles the "edit this page" link when internationalized:
https://github.com/facebook/docusaurus/blob/5c9e4a2dcd12689b84f488a09c6ec6eb6e89531e/website/docusaurus.config.js#L153

I suppose i could i hack my way to a solution, but it would probably be more cost-effective if someone knowledgeable with the data-structure would look at it. (@LeeAlephium @mvaivre? <3 @nop33 is on holidays)

I would do something like this:

```
    [
      'content-docs',
      /** @type {import('@docusaurus/plugin-content-docs').Options} */
      ({
        id: 'community',
        path: 'community',
        routeBasePath: 'community',
        editUrl: ({locale, versionDocsDirPath, docPath}) => {
          if (locale !== 'en') {
            return `https://github.com/alephium/wiki/tree/master//i18n/${locale}/docusaurus-plugin-content-docs/current/${versionDocsDirPath}/${docPath`;
          }
          return `https://github.com/alephium/wiki/tree/master/${versionDocsDirPath}/${docPath}`;
        },
        remarkPlugins: [npm2yarn],
        editCurrentVersion: true,
        sidebarPath: require.resolve('./sidebarsCommunity.js'),
        showLastUpdateAuthor: true,
        showLastUpdateTime: true,
      }),
    ],
```
But i am not entirely sure how/where to place it in the `docusaurus.config.js` :disappointed:

File: README.md (hunk 2 above)
Comment (Sakrecoer) [reply README.md]: added

File: i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md (hunk 7 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md]: fixed

File: i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md (hunk 9 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: i trust you on this :)

Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: There are LOTS of places where we use ```json, though...

Comment (nop33) [reply i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: If you wanna use comments inside the json block, use js. Otherwise json.

Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: ok, but you are suggesting to add js on a json block?

Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/wallet/Wallet-Guide.md]: aaah sorry! i get you now... need more coffee

**Comment by nop33** [comment]
@Sakrecoer I pushed a fix for the "edit this page" link here: https://github.com/alephium/wiki/commit/d7baf2c30c74a5847be14ab582906627ab7c49e4

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md [hunk 11]
```diff
@@ -0,0 +1,55 @@
+---
+sidebar_position : 10
+titre : Guide complet de démarrage du fullnode
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md]: Still needs to be changed.

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md [hunk 12]
```diff
@@ -0,0 +1,161 @@
+---
+sidebar_position : 30
+titre : Full Node sur Raspberry Pi
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md]: here too

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+
+Donc, ci-dessous est la définition du service que vous pouvez mettre dans un fichier `docker-compose.yml`, et simplement appeler `docker-compose up -d` pour démarrer votre noeud complet à partir de cette définition.
+
+```yaml
+version : "3"
+services :
+  broker :
+    image : "alephium/alephium:v1.2.6"
+    restart : unless-stopped
+    ports :
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md]: Here too

File: i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md [hunk 14]
```diff
@@ -0,0 +1,17 @@
+---
+sidebar_position : 20
+title : Sur les poussières d'output et l'explosion des états
```
Comment (nop33) [inline i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md]: here too

**Comment by Sakrecoer** [comment]
> I pushed a fix for the "edit this page" link here: https://github.com/alephium/wiki/commit/d7baf2c30c74a5847be14ab582906627ab7c49e4

You are the BEST @nop33 !!! 

Sorry for missing those other places. Let me address that right away!

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md (hunk 11 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-Node-Starter-Guide.md]: fixed in https://github.com/alephium/wiki/pull/39/commits/aa98c0da5a79c88cbe3ac669aeaa6acb4a6c3f8f

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md (hunk 13 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md]: fixed in https://github.com/alephium/wiki/pull/39/commits/aa98c0da5a79c88cbe3ac669aeaa6acb4a6c3f8f

File: i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md (hunk 12 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/full-node/Full-node-on-raspberry-pi.md]: fixed in https://github.com/alephium/wiki/pull/39/commits/aa98c0da5a79c88cbe3ac669aeaa6acb4a6c3f8f

File: i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md (hunk 14 above)
Comment (Sakrecoer) [reply i18n/fr/docusaurus-plugin-content-docs/current/misc/On-dust-outputs-and-state-explosion.md]: fixed in https://github.com/alephium/wiki/pull/39/commits/aa98c0da5a79c88cbe3ac669aeaa6acb4a6c3f8f

**Comment by Sakrecoer** [comment]
I think we are good to go with this, @nop33 ? Yes, some updates will have to be done in the french part, but it's a good start and the more we wait the greater the chance that we get conflicts, or am i mistaken? I can take care of syncing the changes to the french version next week.

**Review by nop33 — APPROVED** [review]

**Pull request merged**